<?xml version="1.0"?>
<launch>

  
<include file="$(find sequential_map_merging)/launch/rviz_visualisation_mcl.launch"/>
<include file="$(find sequential_map_merging)/launch/empty_world.launch" >
  <arg name="world_name" value="$(find sequential_map_merging)/worlds/map10_world/model.sdf"/>
</include>

<!--#############################################################-->
<group ns="ABot1">
<param name="tf_prefix" value="/ABot1" />
<node pkg="tf" type="static_transform_publisher" name="map_odom_static_broadcaster" args="0 0 0 0 0 0 1 map /ABot1/odom 100" />
<node pkg="tf" type="static_transform_publisher" name="laser" args="0.225 0 0.2 0 0 0 1 /ABot1/base_link /ABot1/base_laser 100" />

  <!-- send ABot urdf to param server -->
  

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_st_pub" >
  </node>
<param name="robot_description1" command="$(find xacro)/xacro '$(find sequential_map_merging)/urdf/ABot.urdf.xacro'" />
  <!-- push ABot_description to factory and spawn robot in gazebo -->
  <node name="spawn_ABot" pkg="gazebo_ros" type="spawn_model" args=" -x 16.2 -y -16.5 -Y 2.88 -urdf -model ABot1 -param robot_description1" respawn="false" output="screen" />


  <node name="joy_node_ABot" pkg="joy" type="joy_node" output="screen">
  <param name="dev" value="/dev/input/js1" />
  <remap from="/joy" to="/ABot/joy"/>
  </node>
  <node name="joystick_controler_ABot" pkg="sequential_map_merging" type="joystick_controler_new.py" output="screen">
  <remap from="/cmd_vel" to="/ABot1/cmd_vel"/>
  <remap from="/joy" to="/ABot1/joy"/>
  </node>

<node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">

<!-- Topic names -->
<param name="scan_topic" value="/ABot1/scan" />

<!-- Frame names -->
<param name="base_frame" value="/ABot1/base_laser" />
<param name="odom_frame" value="/ABot1/odom"/>
<param name="output_timing" value="false"/>

<!-- Tf use -->
<param name="use_tf_scan_transformation" value="true"/>
<param name="use_tf_pose_start_estimate" value="false"/>
<param name="pub_map_odom_transform" value="true"/>
<param name="advertise_map_service" value="true"/>

<!-- Map size / start point -->
<param name="map_resolution" value="0.05"/>
<param name="map_size" value="2048"/>
<param name="map_start_x" value="0.5"/>
<param name="map_start_y" value="0.5" />
<param name="map_multi_res_levels" value="1" />

<!-- Map update parameters -->
<param name="update_factor_free" value="0.4"/>
<param name="update_factor_occupied" value="0.9" />
<param name="map_update_distance_thresh" value="0.1"/>
<param name="map_update_angle_thresh" value="0.06" />
<param name="laser_max_dist" value="30.0" />

<!-- Debug parameters -->
<!--
<param name="pub_drawings" value="true"/>
<param name="pub_debug_output" value="true"/>
-->
</node>
  

</group>

<!--#############################################################-->
<group ns="ABot2">
<param name="tf_prefix" value="/ABot2" />
<node pkg="tf" type="static_transform_publisher" name="map_odom_static_broadcaster" args="0 0 0 0 0 0 1 map /ABot2/odom 100" />
<node pkg="tf" type="static_transform_publisher" name="laser" args="0.225 0 0.2 0 0 0 1 /ABot2/base_link /ABot2/base_laser 100" />

  <!-- send ABot urdf to param server -->
  

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_st_pub" >
  </node>
<param name="robot_description2" command="$(find xacro)/xacro '$(find sequential_map_merging)/urdf/ABot.urdf.xacro'" />
  <!-- push ABot_description to factory and spawn robot in gazebo -->
  <node name="spawn_ABot" pkg="gazebo_ros" type="spawn_model" args=" -x -9.6 -y -15.5 -Y -1.5707 -urdf -model ABot2 -param robot_description2" respawn="false" output="screen" />


  <node name="joy_node_ABot" pkg="joy" type="joy_node" output="screen">
  <param name="dev" value="/dev/input/js0" />
  <remap from="/joy" to="/ABot2/joy"/>
  </node>
  <node name="joystick_controler_ABot" pkg="sequential_map_merging" type="joystick_controler_new.py" output="screen">
  <remap from="/cmd_vel" to="/ABot2/cmd_vel"/>
  <remap from="/joy" to="/ABot2/joy"/>
  </node>

<node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">

<!-- Topic names -->
<param name="scan_topic" value="/ABot2/scan" />

<!-- Frame names -->
<param name="base_frame" value="/ABot2/base_laser" />
<param name="odom_frame" value="/ABot2/odom"/>
<param name="output_timing" value="false"/>

<!-- Tf use -->
<param name="use_tf_scan_transformation" value="true"/>
<param name="use_tf_pose_start_estimate" value="false"/>
<param name="pub_map_odom_transform" value="true"/>
<param name="advertise_map_service" value="true"/>

<!-- Map size / start point -->
<param name="map_resolution" value="0.05"/>
<param name="map_size" value="2048"/>
<param name="map_start_x" value="0.5"/>
<param name="map_start_y" value="0.5" />
<param name="map_multi_res_levels" value="1" />

<!-- Map update parameters -->
<param name="update_factor_free" value="0.4"/>
<param name="update_factor_occupied" value="0.9" />
<param name="map_update_distance_thresh" value="0.1"/>
<param name="map_update_angle_thresh" value="0.06" />
<param name="laser_max_dist" value="30.0" />

<!-- Debug parameters -->
<!--
<param name="pub_drawings" value="true"/>
<param name="pub_debug_output" value="true"/>
-->
</node>
  

</group>



</launch>
